### PR TITLE
Pull package lists into a new view with tabs

### PR DESCRIPTION
**Pull request by cnunciato**
This (admittedly large -- sorry!) change pulls several package-related lists
&mdash; versions, releases and builds, initially &mdash; into tabs on a new
package-level view, and adds a new tab featuring the package's latest release.

Additionally:

* Adds two new Angular modules, PackageModule and SharedModule, for housing
  package-related and general-purpose components and functionality, respectively.

* Adds two placeholder tabs for Build Settings and Readme and suppresses them.

* Adds a new copy-to-clipboard component and pulls channels its own component.

* Removes some dead code, loosens some CSS linting rules and suppresses some
  Sass warnings.

Signed-off-by: Christian Nunciato <cnunciato@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by cnunciato** [comment]
new-tabs

**Comment by cnunciato** [comment]
Resolves #2994.

File: components/builder-web/app/dashboard/_dashboard.component.scss [hunk 1]
```diff
@@ -157,8 +157,8 @@
             }
 
             hab-icon {
-              width: 24px;
-              height: 24px;
+              width: 20px;
+              height: 20px;
```
Comment (mgamini) [inline components/builder-web/app/dashboard/_dashboard.component.scss]: Seems like this should be a var maybe?

File: components/builder-web/app/package/package-builds/_package-builds.component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.hab-package-builds {
```
Comment (mgamini) [inline components/builder-web/app/package/package-builds/_package-builds.component.scss]: Need this?

File: components/builder-web/app/package/package-readme/package-readme.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<div class="hab-package-readme">
+  Soon.
```
Comment (elliott-davis) [inline components/builder-web/app/package/package-readme/package-readme.component.html]: tenor-263043987

File: components/builder-web/app/package/package-sidebar/package-sidebar.component.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    get runCommand() {
+        return `hab start ${this.origin}/${this.name}`;
+    }
+
+    get buildable() {
+      let isMember = !!this.store.getState().origins.mine.find((o) => {
+
+        // Still limiting builds to the core origin
+        return o.name === this.origin && this.origin === "core";
```
Comment (elliott-davis) [inline components/builder-web/app/package/package-sidebar/package-sidebar.component.ts]: Is this also disallowed on the backend?

File: components/builder-web/app/package/package-builds/_package-builds.component.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+
```
Comment (elliott-davis) [inline components/builder-web/app/package/package-builds/_package-builds.component.scss]: Is there a reason this is here and blank?

Comment (cnunciato) [reply components/builder-web/app/package/package-builds/_package-builds.component.scss]: I just wanted to create placeholders for each of these components' styles in a consistent way throughout the module. I suspect these will be used, but as of right now, they aren't strictly necessary.

File: components/builder-web/app/dashboard/_dashboard.component.scss (hunk 1 above)
Comment (cnunciato) [reply components/builder-web/app/dashboard/_dashboard.component.scss]: Yep, good idea. There's a lot we can do to refactor and reduce CSS duplication in general. Might as well start now! 😄

**Comment by cnunciato** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/269749620).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/269749620).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**